src/Data/Binary/Serialise/CBOR/Extra.hs

```diff
@@ -70,3 +83,91 @@ decodeMaybeWith dec = do
       1 -> do !x <- dec
               return (Just x)
       _ -> fail "unknown tag"
+
+--
+-- Encoding and decoding sets
+--
+
+encodeSetLikeWith :: (a -> Encoding) -> [a] -> Encoding
+encodeSetLikeWith f xs = encodeSetSkel f length foldr xs
+
```

Comment:
I'd prefer not to complicate the CBOR encoding/decoding code unless absolutely necessary, and I don't think we need special support for CBOR sets. Let's just lists.